src/garage/tf/models/gaussian_mlp_model2.py

```diff
@@ -0,0 +1,248 @@
+        std_parameterization (str): How the std should be parametrized. There
+            are two options:
+            - exp: the logarithm of the std will be stored, and applied a
+               exponential transformation
+            - softplus: the std will be computed as log(1+exp(x))
+        layer_normalization (bool): Bool for using layer normalization or not.
+
+    """
+
+    # pylint: disable=assignment-from-no-return
+    def __init__(self,
+                 output_dim,
+                 name=None,
+                 hidden_sizes=(32, 32),
+                 hidden_nonlinearity=tf.nn.tanh,
+                 hidden_w_init=tf.glorot_uniform_initializer(),
+                 hidden_b_init=tf.zeros_initializer(),
+                 output_nonlinearity=None,
+                 output_w_init=tf.glorot_uniform_initializer(),
+                 output_b_init=tf.zeros_initializer(),
+                 learn_std=True,
+                 adaptive_std=False,
+                 std_share_network=False,
+                 init_std=1.0,
+                 min_std=1e-6,
+                 max_std=None,
+                 std_hidden_sizes=(32, 32),
+                 std_hidden_nonlinearity=tf.nn.tanh,
+                 std_hidden_w_init=tf.glorot_uniform_initializer(),
+                 std_hidden_b_init=tf.zeros_initializer(),
+                 std_output_nonlinearity=None,
+                 std_output_w_init=tf.glorot_uniform_initializer(),
+                 std_parameterization='exp',
+                 layer_normalization=False):
+        # Network parameters
+        super().__init__(name)
+        self._hidden_sizes = hidden_sizes
+        self._output_dim = output_dim
+        self._learn_std = learn_std
+        self._adaptive_std = adaptive_std
+        self._std_share_network = std_share_network
+        self._std_hidden_sizes = std_hidden_sizes
+        self._min_std = min_std
+        self._max_std = max_std
+        self._std_hidden_nonlinearity = std_hidden_nonlinearity
+        self._std_hidden_w_init = std_hidden_w_init
+        self._std_hidden_b_init = std_hidden_b_init
+        self._std_output_nonlinearity = std_output_nonlinearity
+        self._std_output_w_init = std_output_w_init
+        self._std_parameterization = std_parameterization
+        self._hidden_nonlinearity = hidden_nonlinearity
+        self._hidden_w_init = hidden_w_init
+        self._hidden_b_init = hidden_b_init
+        self._output_nonlinearity = output_nonlinearity
+        self._output_w_init = output_w_init
+        self._output_b_init = output_b_init
+        self._layer_normalization = layer_normalization
+
+        # Tranform std arguments to parameterized space
+        self._init_std_param = None
+        self._min_std_param = None
+        self._max_std_param = None
+        if self._std_parameterization == 'exp':
+            self._init_std_param = np.log(init_std)
+            if min_std is not None:
```

Comment:
looks like the min_std case isn't covered

Reply:
Interesting, I already have a test for min_std
https://github.com/rlworkgroup/garage/blob/new_primitive_for_tfp/tests/garage/tf/models/test_gaussian_mlp_model2.py#L293

Reply:
okay -- perhaps just look at the test report for more info.

Reply:
is it because the default value of `min_std` is not `None`? I already have a test which explicitly set `min_std` to some value.

Reply:
probably, yes. branch coverage measures whether you followed each branch (visited inside each if statement).